app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,85 @@
+class DriversController < ApplicationController
+  def index
+    @drivers = Driver.paginate(page: params[:page]).order("id")
+  end
+
+  def show
+    driver_id = params[:id]
+    @driver = Driver.find_by(id: driver_id)
+    if @driver.nil?
+      flash[:red] = "Driver not found."
+      redirect_to drivers_path
+      return
+    end
+  end
+
+  def new
+    @driver = Driver.new
+  end
+
+  def create
+    @driver = Driver.new(driver_params)
+    if @driver.save
+      flash[:green] = "#{@driver.name} has been successfully created."
+      redirect_to driver_path(@driver)
+      return
+    else
+    render :new
+    return
+    end
+  end
+
+  def edit
+    @driver = Driver.find_by(id: params[:id])
+
+    if @driver.nil?
+      # head :not_found
+      flash[:red] = "Driver not found."
```

Comment:
You should use a more meaningful name than "red" here.  

If the intent is that this is an error you should name the key "error".